src/routes/transactions/transactions.service.ts

```diff
@@ -218,6 +224,72 @@ export class TransactionsService {
     };
   }
 
+  private adjustTransactionHistoryPage(
+    limit?: number,
+    offset?: number,
+  ): PaginationData {
+    if (offset !== undefined && offset > 0) {
+      return new PaginationData(
+        limit ? limit + 1 : PaginationData.DEFAULT_LIMIT + 1,
+        offset - 1,
+      );
+    } else {
+      return new PaginationData(
+        limit ? limit : PaginationData.DEFAULT_LIMIT,
+        PaginationData.DEFAULT_OFFSET,
+      );
+    }
+  }
+
+  async getTransactionHistory(
+    chainId: string,
+    routeUrl: Readonly<URL>,
+    safeAddress: string,
+    timezoneOffset?: string,
+    paginationData?: PaginationData,
```

Comment:
This should not be `undefined`. You should always have a `PaginationData`

Reply:
Done 3ed0214c5e8c5f8593e35c3295694f62ae20da5e